app/com/arpnetworking/metrics/portal/scheduling/JobCoordinator.java

```diff
@@ -0,0 +1,237 @@
+                .setGetPage(offset -> repo.createQuery(_organization)
+                        .offset(offset)
+                        .limit(JOB_QUERY_PAGE_SIZE)
+                        .execute()
+                        .values())
+                .build();
+    }
+
+    private void runAntiEntropy() {
+        final Instant startTime = _clock.instant();
+        System.out.println("in runAntiEntropy");
+        final Iterator<? extends Job<T>> js = getAllJobs();
+        js.forEachRemaining(job -> {
+            System.out.println("considering " + job);
+            final JobRef<T> ref = new JobRef.Builder<T>()
+                    .setRepositoryType(_repositoryType)
+                    .setOrganization(_organization)
+                    .setId(job.getId())
+                    .build();
+            _jobExecutorRegion.tell(
+                    new JobExecutorActor.Reload.Builder<T>()
+                            .setJobRef(ref)
+                            .setETag(job.getETag())
+                            .build(),
+                    getSelf());
+        });
+
+        // We now know that all jobs in the repo have current actors.
+        // There might still be actors which don't correspond to jobs, but that's fine:
+        //   they should self-terminate next time they execute.
+
+        _periodicMetrics.recordTimer(
+                "job_coordinator_tick_time",
+                ChronoUnit.NANOS.between(startTime, _clock.instant()),
+                Optional.of(NANOS));
+        System.out.println("done with runAntiEntropy");
+    }
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .match(AntiEntropyTick.class, message -> {
+
+                    LOGGER.info()
+                            .setMessage("ticking")
+                            .addData("organization", _organization)
+                            .addData("repositoryType", _repositoryType)
+                            .log();
+                    if (_currentlyExecuting) {
+                        return;
+                    }
+
+                    LOGGER.info()
+                            .setMessage("starting anti-entropy")
+                            .addData("organization", _organization)
+                            .addData("repositoryType", _repositoryType)
+                            .log();
+
+                    getContext().getSystem().scheduler().scheduleOnce(
+                            scala.concurrent.duration.Duration.Zero(),
+                            () -> {
+                                try {
+                                    runAntiEntropy();
+                                } finally {
+                                    getSelf().tell(AntiEntropyFinished.INSTANCE, getSelf());
+                                }
+                            },
+                            getContext().getSystem().dispatcher()); // TODO(spencerpearson): probably not the right dispatcher
```

Comment:
@BrandonArp confirm?

Reply:
What is the reason you're kicking this off to a background executor?  Running it on the system dispatcher is likely a bad thing to do.  At least using the context().dispatcher() would let us configure a specific thread pool if we chose to.  Overall, though, I don't like you running this separate from the message processing loop.  For instance, using the _clock variable in the runAntiEntropy() method is already a race condition, and very easy to do by accident in an actor (because you assume that you're already synchronized on the message handling).

Reply:
> What is the reason you're kicking this off to a background executor?
Pretty much just following the "don't block the actor" rule: iterating over all the jobs might take a long time, so I backgrounded it in order to free up the JobCoordinator to handle other messages. (There aren't _currently_ any other messages it might receive, but that could change.)

I hadn't realized that `_clock` and several other instance variables were being referenced asynchronously, but... it seems fine to me? All the instance variables (except `_currentlyExecuting`) are final and (conceptually) immutable.

I guess it does seem a potentially slippery slope. The best alternative I can think of is to have `runAntiEntropy(...)` not reference any instance variables, instead having the necessary objects passed in, with the javadoc explicitly stating that those objects must be (conceptually) immutable. Acceptable?

Reply:
Made `runAntiEntropy` static: https://github.com/ArpNetworking/metrics-portal/pull/162/commits/41c9e4e68bd94649479c6d2e77358e6a89144812
(Also wrote a wrapper method that ensures all instance variables are captured at Runnable-_creation_ time rather than Runnable-_execution_ time.)

Also changed to use `context().dispatcher()`. But I left the thread limit the same, because this actor should only ever have one background task running. The dispatcher that all the _executors_ use, though, might want a higher thread count. Intend to sync with you about that after standup.